From 7b576cdf6f6a0ecbf86893f4b770b42f188ecfde Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:23:42 -0400
Subject: [cpufreq] remove unused notifier CPUFREQ_{SUSPENDCHANGE|RESUMECHANGE}

Message-id: <1409235879-1283-129-git-send-email-prarit@redhat.com>
Patchwork-id: 89518
O-Subject: [RHEL7.1 PATCH BZ 1134639 128/185] cpufreq: remove unused notifier: CPUFREQ_{SUSPENDCHANGE|RESUMECHANGE}
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 0b443ead714f0cba797a7f2476dd756f22b5421e
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Wed Mar 19 11:24:58 2014 +0530

    cpufreq: remove unused notifier: CPUFREQ_{SUSPENDCHANGE|RESUMECHANGE}

    Two cpufreq notifiers CPUFREQ_RESUMECHANGE and CPUFREQ_SUSPENDCHANGE have
    not been used for some time, so remove them to clean up code a bit.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Reviewed-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    [rjw: Changelog]
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/cpu-freq/core.txt b/Documentation/cpu-freq/core.txt
index ce0666e..0060d76 100644
--- a/Documentation/cpu-freq/core.txt
+++ b/Documentation/cpu-freq/core.txt
@@ -92,7 +92,3 @@ values:
 cpu - number of the affected CPU
 old - old frequency
 new - new frequency
-
-If the cpufreq core detects the frequency has changed while the system
-was suspended, these notifiers are called with CPUFREQ_RESUMECHANGE as
-second argument.
diff --git a/arch/powerpc/oprofile/op_model_cell.c b/arch/powerpc/oprofile/op_model_cell.c
index b9589c1..4c5ea12 100644
--- a/arch/powerpc/oprofile/op_model_cell.c
+++ b/arch/powerpc/oprofile/op_model_cell.c
@@ -1122,8 +1122,7 @@ oprof_cpufreq_notify(struct notifier_block *nb, unsigned long val, void *data)
  int ret = 0;
  struct cpufreq_freqs *frq = data;
  if ((val == CPUFREQ_PRECHANGE && frq->old < frq->new) ||
-     (val == CPUFREQ_POSTCHANGE && frq->old > frq->new) ||
-     (val == CPUFREQ_RESUMECHANGE || val == CPUFREQ_SUSPENDCHANGE))
+     (val == CPUFREQ_POSTCHANGE && frq->old > frq->new))
   set_spu_profiling_frequency(frq->new, spu_cycle_reset);
  return ret;
 }
diff --git a/arch/x86/kernel/tsc.c b/arch/x86/kernel/tsc.c
index 930e5d4..e8bb8ce 100644
--- a/arch/x86/kernel/tsc.c
+++ b/arch/x86/kernel/tsc.c
@@ -715,8 +715,7 @@ static int time_cpufreq_notifier(struct notifier_block *nb, unsigned long val,
   tsc_khz_ref = tsc_khz;
  }
  if ((val == CPUFREQ_PRECHANGE  && freq->old < freq->new) ||
-   (val == CPUFREQ_POSTCHANGE && freq->old > freq->new) ||
-   (val == CPUFREQ_RESUMECHANGE)) {
+   (val == CPUFREQ_POSTCHANGE && freq->old > freq->new)) {
   *lpj = cpufreq_scale(loops_per_jiffy_ref, ref_freq, freq->new);
 
   tsc_khz = cpufreq_scale(tsc_khz_ref, ref_freq, freq->new);
diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index d955397..c8a1150 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -264,8 +264,7 @@ static void adjust_jiffies(unsigned long val, struct cpufreq_freqs *ci)
   pr_debug("saving %lu as reference value for loops_per_jiffy; freq is %u kHz\n",
     l_p_j_ref, l_p_j_ref_freq);
  }
- if ((val == CPUFREQ_POSTCHANGE && ci->old != ci->new) ||
-     (val == CPUFREQ_RESUMECHANGE || val == CPUFREQ_SUSPENDCHANGE)) {
+ if (val == CPUFREQ_POSTCHANGE && ci->old != ci->new) {
   loops_per_jiffy = cpufreq_scale(l_p_j_ref, l_p_j_ref_freq,
         ci->new);
   pr_debug("scaling loops_per_jiffy to %lu for frequency %u kHz\n",
diff --git a/include/linux/cpufreq.h b/include/linux/cpufreq.h
index a6a48e4..7cdc028 100644
--- a/include/linux/cpufreq.h
+++ b/include/linux/cpufreq.h
@@ -159,8 +159,6 @@ static inline bool policy_is_shared(struct cpufreq_policy *policy)
 
 #define CPUFREQ_PRECHANGE (0)
 #define CPUFREQ_POSTCHANGE (1)
-#define CPUFREQ_RESUMECHANGE (8)
-#define CPUFREQ_SUSPENDCHANGE (9)
 
 struct cpufreq_freqs {
  unsigned int cpu; /* cpu nr */
-- 
1.7.1